lib/token-range.js

```diff
@@ -0,0 +1,155 @@
+'use strict';
+
+const util = require('util');
+
+/** 
+ * Represents a range of tokens on a Cassandra ring.
+ *
+ * A range is start-exclusive and end-inclusive.  It is empty when
+ * start and end are the same token, except if that is the minimum
+ * token, in which case the range covers the whole ring (this is
+ * consistent with the behavior of CQL range queries).
+ *
+ * Note that CQL does not handle wrapping.  To query all partitions
+ * in a range, see {@link unwrap}.
+ */
+class TokenRange {
+  constructor(start, end, tokenizer) {
+    this.start = start;
+    this.end = end;
+    this._tokenizer = tokenizer;
+  }
+
+  /**
+   * Splits this range into a number of smaller ranges of equal "size"
+   * (referring to the number of tokens, not the actual amount of data).
+   *
+   * Splitting an empty range is not permitted.  But not that, in edge
+   * cases, splitting a range might produce one or more empty ranges.
+   *
+   * @param {Number} numberOfSplits Number of splits to make.
+   * @returns {TokenRage[]} Split ranges.
+   * @throws {Error} If splitting an empty range.
+   */
+  splitEvenly(numberOfSplits) {
```

Comment:
I started by porting all the method definitions from the java driver as a starting point, but I was questioning if it was necessary to implement all of the methods, my thoughts:

1. splitEvently, isEmpty, isWrappedAround, unwrap, contains, equals, compare, toString:  All seem revelant, useful and easy to implement.
2. intersects, intersectWith, mergeWith:  Are useful, but complex.  It is nice for merging/combining ranges (i.e. "Give me a range that applies to replicas X, Y, Z.).  Wasn't sure if I should implement these, but wouldn't take me long.

Reply:
1: definitely +1
2: +1 as it would help on non-trivial table scans

Reply:
- 1: +1
- 2: I think exposing the token info is a good step but it doesn't have to have all the utilities in the java driver, as long as we enable users to create those methods on their own if needed.
I would leave it for the end and if there is enough time, implement them.

Reply:
👍 sounds like a good approach, if have time to get to it i’ll implement them.